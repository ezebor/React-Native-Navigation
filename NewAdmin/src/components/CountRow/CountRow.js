import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import {color} from '../../shared/styles';

const Text = styled.Text`
  font-size: 14px;
  font-weight: normal;
  color: ${color.textGray};
  text-decoration: ${props => (props.textDecoration ? 'line-through' : '')};
`;

const StyledCountRow = styled.View`
  justify-content: center;
  align-items: center;
  border-radius: 4px;
  padding: 4px 16px;
  background-color: ${color.nubeWhite};
`;

const CountRow = ({children, textDecoration}) => {
  return (
    <StyledCountRow>
      <Text textDecoration={textDecoration}>{children}</Text>
    </StyledCountRow>
  );
};

CountRow.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ]).isRequired,
  textDecoration: PropTypes.bool,
};

CountRow.defaultProps = {
  textDecoration: false,
};

export default CountRow;
