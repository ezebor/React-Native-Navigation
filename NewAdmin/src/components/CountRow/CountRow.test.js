import React from 'react';
import renderer from 'react-test-renderer';
import CountRow from './CountRow';

describe('renders correctly CountRow', () => {
  it('Default', () => {
    const tree = renderer.create(<CountRow>8</CountRow>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('With price', () => {
    const tree = renderer.create(<CountRow>$ 10.000</CountRow>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('With textdecoration prop', () => {
    const tree = renderer
      .create(<CountRow textDecoration>$ 10.000</CountRow>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('With long value', () => {
    const tree = renderer
      .create(<CountRow textDecoration>$ 10.000.000.000.000.000.000</CountRow>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
