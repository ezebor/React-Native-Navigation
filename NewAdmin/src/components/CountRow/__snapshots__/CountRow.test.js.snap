// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly CountRow Default 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ecebec",
        "borderRadius": 4,
        "justifyContent": "center",
        "paddingBottom": 4,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 4,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#473c41",
          "fontSize": 14,
          "fontWeight": "normal",
          "textDecorationColor": "black",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        },
      ]
    }
    textDecoration={false}
  >
    8
  </Text>
</View>
`;

exports[`renders correctly CountRow With long value 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ecebec",
        "borderRadius": 4,
        "justifyContent": "center",
        "paddingBottom": 4,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 4,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#473c41",
          "fontSize": 14,
          "fontWeight": "normal",
          "textDecorationColor": "black",
          "textDecorationLine": "line-through",
          "textDecorationStyle": "solid",
        },
      ]
    }
    textDecoration={true}
  >
    $ 10.000.000.000.000.000.000
  </Text>
</View>
`;

exports[`renders correctly CountRow With price 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ecebec",
        "borderRadius": 4,
        "justifyContent": "center",
        "paddingBottom": 4,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 4,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#473c41",
          "fontSize": 14,
          "fontWeight": "normal",
          "textDecorationColor": "black",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        },
      ]
    }
    textDecoration={false}
  >
    $ 10.000
  </Text>
</View>
`;

exports[`renders correctly CountRow With textdecoration prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ecebec",
        "borderRadius": 4,
        "justifyContent": "center",
        "paddingBottom": 4,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 4,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#473c41",
          "fontSize": 14,
          "fontWeight": "normal",
          "textDecorationColor": "black",
          "textDecorationLine": "line-through",
          "textDecorationStyle": "solid",
        },
      ]
    }
    textDecoration={true}
  >
    $ 10.000
  </Text>
</View>
`;
